### PR TITLE
added Jenkinsfile

### PR DESCRIPTION
**Pull request by jasmingacic**
This is just a copy of [Jekinsfile](https://github.com/crossplaneio/stack-gcp/blob/master/Jenkinsfile)

@hasheddan if you want to add anything feel free

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,173 @@
+pipeline {
+    agent { label 'upbound-gce' }
```
Comment (hasheddan) [inline Jenkinsfile:2]: Probably will want to change these to appropriate Packet values.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+        GITHUB_UPBOUND_BOT = credentials('github-upbound-jenkins')
+    }
+
+    stages {
+
+        stage('Prepare') {
+            steps {
+                script {
+                    if (env.CHANGE_ID != null) {
+                        def json = sh (script: "curl -s https://api.github.com/repos/crossplaneio/stack-gcp/pulls/${env.CHANGE_ID}", returnStdout: true).trim()
```
Comment (hasheddan) [inline Jenkinsfile:28]: Update to `packethost/stack-packet`

File: Jenkinsfile [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+        }
+
+                stage("Integration Tests"){
+            when {
+                expression {
+                    return env.shouldBuild != "false"
+                }
+            }
+            steps {
+                sh './build/run make -j\$(nproc) e2e'
```
Comment (hasheddan) [inline Jenkinsfile:98]: Integration test scripts have not been added to this repo, but I can follow up with that. For now, do you want to remove this stage? Example: https://github.com/crossplaneio/stack-gcp/blob/a72c898c8fde9541ed49c5e1e8112e4c92507241/cluster/local/integration_tests.sh

File: Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+                    expression {
+                        return env.shouldBuild != "false"
+                    }
+                }
+            }
+            steps {
+                script {
+                    currentBuild.result = 'SUCCESS'
+                 }
+                step([$class: 'MasterCoverageAction', scmVars: [GIT_URL: env.GIT_URL]])
```
Comment (hasheddan) [inline Jenkinsfile:115]: Hmm not sure if this is configured on this repo...

File: Jenkinsfile [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+        timestamps()
+    }
+
+    environment {
+        RUNNING_IN_CI = 'true'
+        REPOSITORY_NAME = "${env.GIT_URL.tokenize('/')[3].split('\\.')[0]}"
+        REPOSITORY_OWNER = "${env.GIT_URL.tokenize('/')[2]}"
+        DOCKER = credentials('dockerhub-upboundci')
+        AWS = credentials('aws-upbound-bot')
+        GITHUB_UPBOUND_BOT = credentials('github-upbound-jenkins')
```
Comment (hasheddan) [inline Jenkinsfile:19]: This uses an Upbound Bot, so this will likely break on recording test coverage, etc.

**Review by hasheddan — COMMENTED** [review]
There are a number of Crossplane org specific components in this script. It may be worth just setting up a simpler `build` --> `unit tests` --> `push to dockerhub` pipeline to get started.

**Comment by jasmingacic** [comment]
some changes I've added here for testing purposes, I will have them removed

**Pull request merged**